src/routes/reference/basic-reactivity/create-resource.mdx

```diff
@@ -17,206 +17,300 @@ description: >-
-	deferStream?: boolean
-	ssrLoadFrom?: "initial" | "server"
-	storage?: (
-		init: T | undefined
-	) => [Accessor<T | undefined>, Setter<T | undefined>]
-	onHydrated?: (k: S | undefined, info: { value: T | undefined }) => void
-}
+// user() is never undefined
+console.log(user().name); // "Loading..." initially
+```
 
-function createResource<T, U = true>(
-	fetcher: (
-		k: U,
-		info: { value: T | undefined; refetching: boolean | unknown }
-	) => T | Promise<T>,
-	options?: ResourceOptions<T, U>
-): ResourceReturn<T>
-
-function createResource<T, U>(
-	source: U | false | null | (() => U | false | null),
-	fetcher: (
-		k: U,
-		info: { value: T | undefined; refetching: boolean | unknown }
-	) => T | Promise<T>,
-	options?: ResourceOptions<T, U>
-): ResourceReturn<T>
+### Conditional Fetching
 
+```typescript
+const [isEnabled, setIsEnabled] = createSignal(false);
+
+const [data] = createResource(
+  () => isEnabled() && userId(), // Only fetches when enabled and userId exists
+  async (id) => fetchUserData(id)
+);
 ```
+
+## Details
+
+Resources provide a reactive pattern for handling asynchronous data that integrates seamlessly with SolidJS's reactivity system and Suspense boundaries. 
+Unlike signals, resources are specifically designed for async operations and provide built-in loading states, error handling, and refetching capabilities.
+
+### State Management
+
+State follows a predictable lifecycle: `"unresolved" → "pending" → "ready"/"errored"`. 
+During refetching, the state transitions to `"refreshing"` while maintaining the previous value accessible via the latest property. 
+This prevents UI flickering during updates.
+
+In version v.1.5.0, resources gained the ability to cover a more detailed view of the Resource state, beyond `loading` and `error` states. 
+This includes distinct states for initial loading, successful fetches, refetching, and errors:
+
+| State | Description | Loading | Error | Latest |
+| ------ | ----------- | ------- | ----- | ------ |
+| `unresolved` | Initial state, not yet fetched | `false` | `undefined` | `undefined` |
+| `pending` | Fetching in progress | `true` | `undefined` | `undefined` |
+| `ready` | Successfully fetched | `false` | `undefined` | `T` |
+| `refreshing` | Refetching while keeping previous value | `true` | `undefined` | `T` |
+| `errored` | Fetching failed | `false` | `any` | `undefined` |
+
+### Source Reactivity
+
+Source automatically triggers refetching when the source value changes.
+The fetcher only runs when the source is truthy (not `null`, `undefined`, or `false`), allowing for conditional fetching patterns.
+Sources can be static values, reactive accessors, or computed values.
+
+### Refetching Behavior
+
+Refetching lets you manually trigger the fetcher using the `refetch` action, optionally passing extra parameters for more control over the request. 
+The `mutate` action allows you to optimistically update the resource value locally, without making a network request.
```

Comment:
this is interesting to know. I don't think I've used the mutate action at all. Good call out